### PR TITLE
replacement postgresql role

### PR DESCRIPTION
**Pull request by kayiwa**
this role combines the db and user management

### CODE REVIEW

File: roles/pulibrary.approvals/meta/main.yml [hunk 1]
```diff
@@ -16,4 +16,21 @@ dependencies:
   - role: 'pulibrary.rails-app'
   - role: 'pulibrary.mailcatcher'
   - role: 'pulibrary.postgresql'
-  - role: 'pulibrary.samba'
\ No newline at end of file
+    vars:
```
Comment (carolyncole) [inline roles/pulibrary.approvals/meta/main.yml:19]: @kayiwa why not put these in ` roles/pulibrary.approvals/vars/default.yml`?  I would never look for vars here.  Is this a new best practice?

Comment (kayiwa) [reply roles/pulibrary.approvals/meta/main.yml:19]: It isn't baked yet but yes @carolyncole I can't recall anymore but I think I tried it failed.

File: roles/pulibrary.psql/molecule/default/tests/test_default.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    assert pkg.is_installed
+
+
+def test_postgresql_server_listening_port(host):
+    socket = host.socket("tcp://127.0.0.1:5432")
+
+    assert socket.is_listening
+
+
+def test_for_repmgr_config_file(host):
```
Comment (carolyncole) [inline roles/pulibrary.psql/molecule/default/tests/test_default.py:27]: Would be better to add test for database connection.

File: roles/pulibrary.psql/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
+---
+- name: PostgreSQL | Make sure the CA certificates are available
+  apt:
+    pkg: ca-certificates
+    state: present
+
+- name: pgbouncer | Add PostgreSQL repository apt-key | apt
```
Comment (carolyncole) [inline roles/pulibrary.psql/tasks/main.yml:7]: Should be postgres

File: roles/pulibrary.psql/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  apt_key:
+    id: "{{ postgresql_apt_key_id }}"
+    url: "{{ postgresql_apt_key_url }}"
+    state: present
+
+- name: pgbouncer | Add PostgreSQL repository | apt
```
Comment (carolyncole) [inline roles/pulibrary.psql/tasks/main.yml:13]: Should say postgres

File: roles/pulibrary.psql/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+- name: Postgresql | install postgresql server packages
+  apt:
+    name: '{{ item }}'
+    state: present
+    update_cache: true
+  loop:
+    - "{{ postgresql_packages }}"
+
+- name: Postgresql | install postgresql client packages
```
Comment (carolyncole) [inline roles/pulibrary.psql/tasks/main.yml:26]: Where and when does this get installed for user of this role?  Most the rails package installs these dependencies, may this is a moot point.

File: roles/pulibrary.psql/tasks/users.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    port: "{{ postgresql_port }}"
+    state: present
+    login_user: "{{ postgresql_admin_user }}"
+  no_log: false
+  become: true
+  become_user: "{{ postgresql_admin_user }}"
+  with_items: "{{ postgresql_users }}"
+  when: postgresql_users|length > 0
+
+- name: PostgreSQL | Update the user privileges
```
Comment (carolyncole) [inline roles/pulibrary.psql/tasks/users.yml:21]: Could this be combined with the above step?

Comment (carolyncole) [reply roles/pulibrary.psql/tasks/users.yml:21]: Add 26 28 and 30 to above

File: roles/pulibrary.pulfalight/molecule/default/playbook.yml [hunk 7]
```diff
@@ -5,6 +5,7 @@
     - role: pulibrary.pulfalight
   vars:
     - running_on_server: false
+    - node_identifier: 1
```
Comment (carolyncole) [inline roles/pulibrary.pulfalight/molecule/default/playbook.yml:8]: Can be removed

File: playbooks/postgresql_staging.yml [hunk 8]
```diff
@@ -1,13 +1,27 @@
...
+    - role: ../roles/pulibrary.psql
+
+
+- hosts: pgbouncer_staging
+  remote_user: pulsys
+  become: true
+  vars_files:
+    - ../group_vars/postgresql/staging.yml
+    - ../group_vars/postgresql/pg_bouncer_staging.yml
+    - ../host_vars/lib-postgres-staging1.princeton.edu.yml
```
[the remark this answers is not in the document]
Comment (kayiwa) [reply playbooks/postgresql_staging.yml:20]: It does AFAICT

File: roles/pulibrary.pgbouncer/molecule/default/playbook.yml [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+            - gis
+      postgresql_users:
+        - name: app_user
+          password: "ToughToGusspAsSwOrD"
+          db: awesome_db
+        - name: pgbouncer_user
+          password: "ToughToGusspAsSwOrD"
+          db: test_db
+        - name: replicant
+          password: "VoightKampff"
```
Comment (jrgriffiniii) [inline roles/pulibrary.pgbouncer/molecule/default/playbook.yml:30]: 🦄

**Pull request merged**